src/index_notation/transformations.cpp

```diff
@@ -1487,6 +1492,310 @@ ir::Stmt GEMM::replaceValidStmt(IndexStmt stmt,
+  }
+
+  results.push_back(ir::VarDecl::make(
+      cStrides,
+      ir::Call::make(
+          "std::vector<tblis::stride_type>", {cTensorVar.getOrder()}, Auto)));
+  for (int i = 0; i < cTensorVar.getOrder(); ++i) {
+    results.push_back(ir::Store::make(cStrides, i, getStride(cAccess, i)));
+  }
+
+  // Create TBLIS tensors.
+  Datatype TblisTensor("tblis::tblis_tensor");
+  auto aTensor = ir::Var::make("aTensor", TblisTensor);
+  auto bTensor = ir::Var::make("bTensor", TblisTensor);
+  auto cTensor = ir::Var::make("cTensor", TblisTensor);
+
+  std::vector<ir::Expr> aArgs = {
+    ir::Call::make("&", {aTensor}, Auto),
+    aTensorVar.getOrder(), // ndim
+    ir::MethodCall::make(aShape, "data", {}, false /* deref */, Auto), // shape
+    getPtr(aAccess, aDomain), // data
+    ir::MethodCall::make(aStrides, "data", {}, false /* deref */, Auto), // strides
+  };
+  results.push_back(ir::VarDecl::make(aTensor, ir::Symbol::make("{}")));
+  // TODO (rohany): Pick the right call between double and float here.
+  results.push_back(ir::SideEffect::make(ir::Call::make("tblis::tblis_init_tensor_d", aArgs, Auto)));
+
+  std::vector<ir::Expr> bArgs = {
+    ir::Call::make("&", {bTensor}, Auto),
+    bTensorVar.getOrder(), // ndim
+    ir::MethodCall::make(bShape, "data", {}, false /* deref */, Auto), // shape
+    ir::Call::make(unconst_cast, {getPtr(bAccess, aDomain)}, Auto), // data
+    ir::MethodCall::make(bStrides, "data", {}, false /* deref */, Auto), // strides
+  };
+  results.push_back(ir::VarDecl::make(bTensor, ir::Symbol::make("{}")));
+  // TODO (rohany): Pick the right call between double and float here.
+  results.push_back(ir::SideEffect::make(ir::Call::make("tblis::tblis_init_tensor_d", bArgs, Auto)));
+
+  std::vector<ir::Expr> cArgs = {
+    ir::Call::make("&", {cTensor}, Auto),
+    cTensorVar.getOrder(), // ndim
+    ir::MethodCall::make(cShape, "data", {}, false /* deref */, Auto), // shape
+    ir::Call::make(unconst_cast, {getPtr(cAccess, aDomain)}, Auto), // data
+    ir::MethodCall::make(cStrides, "data", {}, false /* deref */, Auto), // strides
+  };
+  results.push_back(ir::VarDecl::make(cTensor, ir::Symbol::make("{}")));
+  // TODO (rohany): Pick the right call between double and float here.
+  results.push_back(ir::SideEffect::make(ir::Call::make("tblis::tblis_init_tensor_d", cArgs, Auto)));
+
+  std::vector<ir::Expr> args = {
+      ir::Symbol::make("tblis_single"),
+      ir::Symbol::make("nullptr"), // tblis_config
+
+      ir::Call::make("&", {bTensor}, Auto),
+      ir::Symbol::make("\"" + this->content->bModes + "\""), // bmodes
+      ir::Call::make("&", {cTensor}, Auto),
+      ir::Symbol::make("\"" + this->content->cModes + "\""), // cmodes
+      ir::Call::make("&", {aTensor}, Auto),
+      ir::Symbol::make("\"" + this->content->aModes + "\""), // amodes
+  };
+
+  results.push_back(ir::SideEffect::make(ir::Call::make("tblis::tblis_tensor_mult", args, Auto)));
```

Comment:
Is there any error checking that we should do on this result?

Reply:
Not really. All the TBLIS functions (including tensor init functions) return void.